rma/models/stock_move.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    # RMAs that were created from the delivery move
+    rma_ids = fields.One2many(
+        comodel_name='rma',
+        inverse_name='move_id',
+        string='RMAs',
+        copy=False,
+    )
+    # RMAs linked to the incoming movement from client
+    rma_receiver_ids = fields.One2many(
+        comodel_name='rma',
+        inverse_name='reception_move_id',
+        string='RMA receivers',
+        copy=False,
+    )
+    # RMA that create the delivery movement to the customer
+    rma_id = fields.Many2one(
+        comodel_name='rma',
+        string='RMA return',
+        copy=False,
+    )
+
+    def unlink(self):
+        if self.mapped('rma_receiver_ids'):
+            raise ValidationError(
+                _("There is at least one move you are trying to "
+                  "delete that is a reception of one or more RMAs."))
+        return super().unlink()
+
+    def _action_done(self):
```

Comment:
When we cancel the move, we should change the state of the rma as well, allowing to make a new reception.